src/app.rb

```diff
@@ -0,0 +1,168 @@
+  end
+
+  # Create a person...
+  def create_person
+    print 'Create a Student (1) or a Teacher (2)? [Input the desired number]: '
+    option = gets.chomp
+    case option
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Invalid option.'
+    end
+  end
+
+  # Create a student...
+  def create_student
+    print 'Type Age: '
+    age = gets.chomp.to_i
+    print 'Type Name: '
+    name = gets.chomp
+    print 'Do you have parent permission? [Yes/No]: '
+    permission = gets.chomp.downcase
+    @people << if permission == 'yes'
+                 Student.new(age, name: name, parent_permission: true)
+               else
+                 Student.new(age, name: name, parent_permission: false)
+               end
+    puts 'Student has been created successfully'
+  end
+
+  def create_teacher
+    print 'Type Age: '
+    age = gets.chomp.to_i
+    print 'Type Name: '
+    name = gets.chomp
+    print 'Specialization: '
+    specialization = gets.chomp
+    @people << Teacher.new(age, name: name, specialization: specialization)
+    puts 'Teacher has been created successfully'
+  end
+
+  # Create a book...
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    @books << Book.new(title, author)
+    puts 'Book has been created successfully'
+  end
+
+  # create rental.....................
+  def create_rental
+    book = select_book
+    person = select_person
+
+    if book && person
+      print 'Date (YYYY/MM/DD): '
+      date = gets.chomp
+
+      @rentals << Rental.new(date, book, person)
+      puts 'Rental has been created successfully'
+    else
+      puts 'Error: You have made an invalid book or person selection.'
+    end
+  end
+
+  # Helper method to select a book...
+  def select_book
+    puts 'Please select a book from the following list by number'
+    @books.each_with_index do |book, index|
+      puts "#{index}) Title: \"#{book.title}\", Author: #{book.author}"
+    end
+
+    book_index = gets.chomp.to_i
+
+    @books[book_index] if (0...@books.length).cover?(book_index)
+  end
+
+  # Helper method to select a person...
+  def select_person
+    puts 'Please select a person from the following list by number (not ID)'
+    @people.each_with_index do |person, index|
+      if person.is_a?(Student)
+        puts "#{index}) [Student] Name: \"#{person.name}\", ID: #{person.id}, Age: #{person.age}"
+      elsif person.is_a?(Teacher)
+        puts "#{index}) [Teacher] Name: \"#{person.name}\", ID: #{person.id}, Age: #{person.age}"
+      end
+    end
+
+    person_index = gets.chomp.to_i
+
+    @people[person_index] if (0...@people.length).cover?(person_index)
+  end
+
+  # List all rentals for a given person ID
+  def list_rentals_for_person
+    print 'Enter the ID of the person: '
+    person_id = gets.chomp.to_i
+
+    person = @people.find { |p| p.id == person_id }
+
```

Comment:
- [ ] There are already created rentals as well for your application which is not supposed to be so, rentals should only be added after creating new users and books. :-1: :sparkles: 
![4](https://github.com/Kaiserabbas/school_library/assets/89055502/41e23fa8-b736-4a91-ad4a-9d766001b041)

Reply:
[Update the code by removing initial data supplied to the app](https://github.com/Kaiserabbas/school_library/pull/4/commits/5fb09cf98f3d48509812bb27f0ce5dbc12be1d40)